packages/common.vm/tools/vm.common/vm.common.psm1

```diff
@@ -886,3 +886,379 @@ RAW_TOOLS_DIR: ${Env:RAW_TOOLS_DIR}
+
+        }
+        else {
+            $validatedData = $data
+        }
+
+        # check if path exists. If not, create the path for the registry value
+        if (!(Test-Path -Path $path)) {
+            # Create the registry key
+            New-Item -Path $path -Force | Out-Null
+            VM-Write-Log "INFO" "`t[+] Registry key created: $path"
+        }
+        else {
+            VM-Write-Log "WARN" "`t[+] Registry key already exists: $path"
+        }
+
+        Set-ItemProperty -Path $path -Name $value -Value $validatedData -Type $type -Force | Out-Null
+        VM-Write-Log "INFO" "[+] $name has been successful"
+    }
+    catch {
+        VM-Write-Log "ERROR" "Failed to update the registry value. Error: $_"
+    }
+}
+
+function VM-Remove-Path {
+# Function for removing Paths/Programs
+    param(
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$name,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateSet("file", "dir")]
+        [string]$type,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$path
+    )
+
+    try {
+        if ($type -eq "file") {
+            if (Test-Path -Path $path -PathType Leaf) {
+                Remove-Item -Path $path -Force
+                VM-Write-Log "INFO" "[+] $name has been successfully removed."
+            } else {
+                VM-Write-Log "WARN" "[+] $path does not exist."
+            }
+        }
+        elseif ($type -eq "dir") {
+            if (Test-Path -Path $path -PathType Container) {
+                Remove-Item -Path $path -Recurse -Force
+                VM-Write-Log "INFO" "[+] $name has been successfully removed."
+            } else {
+                VM-Write-Log "WARN" "[+] $path does not exist."
+            }
+        }
+    }
+    catch {
+        VM-Write-Log "ERROR" "An error occurred while removing the $type $path. Error: $_"
+    }
+}
+
+function VM-Execute-Custom-Command{
+# Function for removing items in need of custom code.
+    param(
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$name,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string[]]$cmds
+    )
+
+    try {
+        Write-Output "[+] Executing commands for '$name':"
+        foreach ($cmd in $cmds) {
+            Write-Output "`t[+] Executing command: $cmd"
+            start-process powershell -ArgumentList "-WindowStyle","Hidden","-Command",$cmd -Wait
+            Write-Host "`t[+] Process completed. Moving to next."
+        }
+        Write-Output "[+] All commands for '$name' have been executed successfully."
+    }
+    catch {
+        VM-Write-Log "ERROR" "An error occurred while executing commands for '$name'. Error: $_"
+    }
+}
+
+function VM-Configure-Prompts {
+    # $Env:MandiantVM must be set in the install script
```

Comment:
Thoughts on remaining to "$Env:VMPrompt"? I think this will keep it company agnostic and be more descriptive

Reply:
the `$Env:MandiantVM` variable is used a few different times, and should be set in the installer script. For instance, Commando's is set to `Commando VM`:
` [Environment]::SetEnvironmentVariable('MandiantVM', 'Commando VM', [EnvironmentVariableTarget]::Machine)`
https://github.com/mandiant/commando-vm/pull/306/files#diff-50d51a233d8ff4886061fc5cf1a03642de1f64b3625903e4f876e8cd2a3f5806

Reply:
lol thoughts on renaming it in both places? I'm pro descriptive variable names and to me the "MandiantVM" is ambiguous.